utils/orchestrator/ecsutil.py

```diff
@@ -0,0 +1,138 @@
+# (C) Datadog, Inc. 2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import logging
+import requests
+import socket
+
+# project
+from utils.dockerutil import DockerUtil
+from utils.platform import Platform
+from utils.singleton import Singleton
+
+
+log = logging.getLogger(__name__)
+
+ECS_INTROSPECT_DEFAULT_PORT = 51678
+ECS_AGENT_CONTAINER_NAME = 'ecs-agent'
+
+class ECSUtil:
+    __metaclass__ = Singleton
+
+    def __init__(self):
+        self.docker_util = DockerUtil()
+        self.ecs_agent_local = None
+
+        self.ecs_tags = {}
+        self._populate_ecs_tags()
+
+    def _get_ecs_address(self):
+        """Detect how to connect to the ecs-agent"""
+        ecs_config = self.docker_util.inspect_container('ecs-agent')
+        ip = ecs_config.get('NetworkSettings', {}).get('IPAddress')
+        ports = ecs_config.get('NetworkSettings', {}).get('Ports')
+        port = ports.keys()[0].split('/')[0] if ports else None
+        if not ip:
+            port = ECS_INTROSPECT_DEFAULT_PORT
+            if self._is_ecs_agent_local():
+                ip = "localhost"
+            elif Platform.is_containerized() and self.docker_gateway:
+                ip = self.docker_gateway
+            else:
+                raise Exception("Unable to determine ecs-agent IP address")
+
+        return ip, port
+
+    def _populate_ecs_tags(self):
+        try:
+            ip, port = self._get_ecs_address()
+        except Exception as ex:
+            self.log.warning("Failed to connect to ecs-agent, skipping task tagging: %s" % ex)
+            return
+
+        try:
+            tasks = requests.get('http://%s:%s/v1/tasks' % (ip, port)).json()
+            for task in tasks.get('Tasks', []):
+                for container in task.get('Containers', []):
+                    tags = ['task_name:%s' % task['Family'], 'task_version:%s' % task['Version']]
+                    self.ecs_tags[container['DockerId']] = tags
+        except requests.exceptions.HTTPError as ex:
+            self.log.warning("Unable to collect ECS task names: %s" % ex)
+
+    def get_container_tags(self, cid):
```

Comment:
Can you please prefix is as private _get_container_tags, to make clear the entrypoint is extract_container_tags?
The issue is kubeutil has get_kube_labels that calls extract_kube_labels, one could get confused.

Or we could directly go with _get_cacheable_tags / _get_uncacheable_tags as we'll refactor the cache lookup logic in extract_container_tags for 5.15